src/components/Apps/AppItem.jsx

```diff
@@ -58,18 +58,27 @@ export class AppItem extends React.Component {
     }
   }
 
+  onAppSwitch = () => {
+    const { onAppSwitch } = this.props
+    if (typeof onAppSwitch === 'function') {
+      setTimeout(() => {
```

Comment:
You should store this timeout and cleat it on willUnmount(). 

It'll prevent possible memory leaks and action on a non mounted component